packages/angular_devkit/architect/src/architect.ts

```diff
@@ -0,0 +1,262 @@
+    );
+  }
+
+  loadWorkspaceFromJson(json: Workspace) {
+    return this._validateAgainstSchema(json, this._workspaceSchema).pipe(
+      concatMap((validatedWorkspace: Workspace) => {
+        this._workspace = validatedWorkspace;
+
+        return of(this);
+      }),
+    );
+  }
+
+  getTarget<OptionsT>(
+    projectName?: string,
+    targetName?: string,
+    configurationName?: string,
+    additionalInput?: Partial<OptionsT>,
+  ): Target<OptionsT> {
+    if (!this._workspace) {
+      throw new WorkspaceNotYetLoadedException();
+    }
+
+    projectName = projectName || this._workspace.defaultProject as string;
+    const project = this._workspace.projects[projectName];
+
+    if (!project) {
+      throw new ProjectNotFoundException(projectName);
+    }
+
+    targetName = targetName || project.defaultTarget as string;
+    const projectTarget = project.targets[targetName];
+
+    if (!projectTarget) {
+      throw new TargetNotFoundException(targetName);
+    }
+
+    const options = projectTarget.options;
+    let configuration;
+
+    if (configurationName) {
+      configuration = projectTarget.configurations
+        && projectTarget.configurations[configurationName];
+
+      if (!configuration) {
+        throw new ConfigurationNotFoundException(configurationName);
+      }
+    }
+
+    // Resolve root for the target.
+    // TODO: add Path format to JSON schemas
+    const target: Target<OptionsT> = {
+      root: resolve(this._root, normalize(project.root)),
+      projectType: project.projectType,
+      builder: projectTarget.builder,
+      options: {
+        ...options,
+        ...configuration ? configuration : {},
+        ...additionalInput ? additionalInput as {} : {},
+      } as OptionsT,
+    };
+
+    return target;
+  }
+
+  // Will run the target using the target.
+  run<OptionsT>(
+    target: Target<OptionsT>, logger?: logging.Logger, stop?: Observable<void>,
+  ): Observable<BuildEvent> {
+    logger = logger || createConsoleLogger();
+    const context = { logger, architect: this, host: this._host };
+
+    let builderDescription: BuilderDescription;
+
+    return this.getBuilderDescription(target).pipe(
+      concatMap(description => {
+        builderDescription = description;
+
+        return this.validateBuilderOptions(target, builderDescription);
+      }),
+      concatMap(() => of(this.getBuilder(builderDescription, context))),
+      concatMap(builder => builder.run(target, stop)),
+    );
+  }
+
+  getBuilderDescription<OptionsT>(target: Target<OptionsT>): Observable<BuilderDescription> {
+    return new Observable((obs) => {
+      // TODO: this probably needs to be more like NodeModulesEngineHost.
+      const basedir = getSystemPath(this._root);
+      const [pkg, builderName] = target.builder.split(':');
+      const pkgJsonPath = nodeResolve(pkg, { basedir, resolvePackageJson: true });
+
+      // Load builders json.
+      const pkgJsonBuildersPath = require(pkgJsonPath)['builders'];
```

Comment:
nit. `builders` as a plural would generally imply an array or object map property.  This may cause some confusion.

Reply:
I agree with you there, but I'm not sure what a better name would be. Schematics also does it similarly with the `schematics` property in `collection.json`.

Reply:
`targets`?

Reply:
Please not another meaning for the word target :(